mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -660,6 +661,8 @@ void LaunchOp::build(OpBuilder &builder, OperationState &result,
   // Add grid and block sizes as op operands, followed by the data operands.
   result.addOperands({gridSizeX, gridSizeY, gridSizeZ, getBlockSizeX,
                       getBlockSizeY, getBlockSizeZ});
+  if (clusterSizeX && clusterSizeY && clusterSizeZ)
```

Comment:
If only one is passed in... it seems wrong to ignore them, I think? Perhaps it would be better to construct the op as given and let the verifier fail?

Reply:
This doesn't seem resolved yet? I would change the condition to use `||` instead of `&&`